### PR TITLE
Let's upgrade to dnx!

### PR DESCRIPTION
**Pull request by Yantrio**
DNX

### CODE REVIEW

**Comment by Yantrio** [comment]
Builds currently fail , but work on my machine. People are welcome to look at them whilst I sleep, otherwise I'll look at them tomorrow :)

File: .settings/launch.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (davidfowl) [inline .settings/launch.json]: No tabs please

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 language: csharp
 sudo: false  # use the new container-based infrastructure
-install: curl -sSL https://raw.githubusercontent.com/aspnet/Home/release/kvminstall.sh | sh && source ~/.k/kvm/kvm.sh && kvm upgrade
+install: curl curl -sSL https://raw.githubusercontent.com/aspnet/Home/dev/dnvminstall.sh | sh && source ~/.dnx/dnvm/dnvm.sh && dnvm upgrade
```
Comment (davidfowl) [inline .travis.yml]: curl curl? Did you look at this :smile:

File: .settings/launch.json (hunk 1 above)
Comment (davidfowl) [inline .settings/launch.json]: Thus file also looks useless. Get rid of it.

File: build.cmd [hunk 3]
```diff
@@ -1,31 +1,31 @@
 @echo off
 
 pushd %~dp0
-set "KRE_NUGET_API_URL=https://www.nuget.org/api/v2"
+set "DNX_NUGET_API_URL=https://www.nuget.org/api/v2"
 setlocal EnableDelayedExpansion 
-where kvm
+where dnx
```
Comment (davidfowl) [inline build.cmd]: dnvm not dnx

File: build.sh [hunk 4]
```diff
@@ -1,24 +1,24 @@
 #!/bin/bash
 rm -rf artifacts
 if ! type kvm > /dev/null 2>&1; then
```
Comment (davidfowl) [inline build.sh]: type dnvm

File: build.sh [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
-    curl -sSL https://raw.githubusercontent.com/aspnet/Home/release/kvminstall.sh | sh && source ~/.k/kvm/kvm.sh
+    curl -sSL https://raw.githubusercontent.com/aspnet/Home/dev/dnvminstall.sh | DNX_BRANCH=dev sh && source ~/.dnx/dnvm/dnvm.sh
 fi
-export KRE_FEED=https://www.nuget.org/api/v2
-kvm install 1.0.0-beta3
-kpm restore
+export DNX_FEED=https://www.nuget.org/api/v2
```
Comment (davidfowl) [inline build.sh]: This isn't required anymore

File: build.sh [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
 cd ../../
-kvm use 1.0.0-beta3
-kpm bundle src/OmniSharp --no-source --out artifacts/build/omnisharp --runtime kre-mono.1.0.0-beta3 2>&1 | tee buildlog
+dnvm use 1.0.0-beta4
+dnu package src/OmniSharp --no-source --out artifacts/build/omnisharp --runtime dnx-mono.1.0.0-beta4 2>&1 | tee buildlog
 # work around for kpm bundle returning an exit code 0 on failure 
 grep "Build failed" buildlog
 rc=$?; if [[ $rc == 0 ]]; then exit 1; fi
 
 # work around for kpm pack not preserving the executable flag on klr when copied
```
Comment (davidfowl) [inline build.sh]: This shouldn't be required anymore

File: scripts/Omnisharp [hunk 7]
```diff
@@ -9,5 +9,5 @@ done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
 
-export KRE_APPBASE=$DIR/../src/OmniSharp
-$HOME/.k/runtimes/kre-mono.1.0.0-beta3/bin/k run "$@"
+export DNX_APPBASE=$DIR/../src/OmniSharp
```
Comment (davidfowl) [inline scripts/Omnisharp]: This shouldn't be required anymore you pass the path to dnx now

File: scripts/Omnisharp [hunk 8]
```diff
@@ -9,5 +9,5 @@ done
...
+$HOME/.dnx/runtimes/dnx-mono.1.0.0-beta4/bin/dnx run "$@"
```
Comment (davidfowl) [inline scripts/Omnisharp]: This is broken dnx {path} run

File: scripts/Omnisharp.cmd [hunk 9]
```diff
@@ -1,4 +1,4 @@
 SETLOCAL
 
-SET "KRE_APPBASE=%~dp0..\src\Omnisharp"
-%USERPROFILE%\.k\runtimes\kre-clr-win-x86.1.0.0-beta3\bin\k run %*
+SET "DNX_APPBASE=%~dp0..\src\Omnisharp"
+%USERPROFILE%\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta4\bin\dnx run %*
```
Comment (davidfowl) [inline scripts/Omnisharp.cmd]: dnx {path} run

File: scripts/Omnisharp.cmd [hunk 10]
```diff
@@ -1,4 +1,4 @@
 SETLOCAL
 
-SET "KRE_APPBASE=%~dp0..\src\Omnisharp"
-%USERPROFILE%\.k\runtimes\kre-clr-win-x86.1.0.0-beta3\bin\k run %*
+SET "DNX_APPBASE=%~dp0..\src\Omnisharp"
```
Comment (davidfowl) [inline scripts/Omnisharp.cmd]: Remove this

File: src/OmniSharp/AspNet5/AspNet5Paths.cs [hunk 11]
```diff
@@ -30,7 +30,7 @@ public class AspNet5Paths
         {
             _env = env;
             _options = options;
-            _logger = loggerFactory.Create<AspNet5Paths>();
+            _logger = loggerFactory.CreateLogger(typeof(AspNet5Paths).FullName);
```
Comment (davidfowl) [inline src/OmniSharp/AspNet5/AspNet5Paths.cs]: `CreateLogger<AspNet5Paths>()`

File: src/OmniSharp/AspNet5/AspNet5Paths.cs [hunk 12]
```diff
@@ -61,7 +61,7 @@ private AspNet5RuntimePathResult GetRuntimePath()
 
                     if (Directory.Exists(path))
                     {
-                        _logger.WriteInformation(string.Format("Using runtime '{0}'.", path));
+                        _logger.LogInformation(string.Format("Using runtime '{0}'.", path));
```
Comment (davidfowl) [inline src/OmniSharp/AspNet5/AspNet5Paths.cs]: File an issue to remove the string.Format calls.

File: src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs [hunk 13]
```diff
@@ -51,7 +51,7 @@ public class AspNet5ProjectSystem : IProjectSystem
         {
             _workspace = workspace;
             _env = env;
-            _logger = loggerFactory.Create<AspNet5ProjectSystem>();
+            _logger = loggerFactory.CreateLogger(typeof(AspNet5ProjectSystem).FullName);
```
Comment (davidfowl) [inline src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs]: The generic overload still exists

File: src/OmniSharp/Program.cs [hunk 14]
```diff
@@ -74,41 +73,40 @@ public void Main(string[] args)
...
             };
+
+            context.Services.AddInstance<IOmnisharpEnvironment>(Environment);
+
+            var serverShutdown = engine.Start(context);
+            var hostingEnv = context.ApplicationServices.GetRequiredService<IHostingEnvironment>();
+            var appEnv = context.ApplicationServices.GetRequiredService<IApplicationEnvironment>();
+            
+            //is this loaded too late? do I need to get it somewhere else
+            context.ApplicationName = appEnv.ApplicationName;
```
Comment (davidfowl) [inline src/OmniSharp/Program.cs]: Why are you setting this after starting the server?

File: tests/OmniSharp.Tests/CodeActionsFacts.cs [hunk 15]
```diff
@@ -1,4 +1,4 @@
-#if ASPNET50
+#if dnx451
```
Comment (davidfowl) [inline tests/OmniSharp.Tests/CodeActionsFacts.cs]: lowercase?

File: tests/OmniSharp.Tests/Fakes/FakeLogger.cs [hunk 16]
```diff
@@ -0,0 +1,20 @@
+using System;
+using Microsoft.Framework.Logging;
+
+public class FakeLogger : ILogger
```
Comment (davidfowl) [inline tests/OmniSharp.Tests/Fakes/FakeLogger.cs]: Uhhhh where is the namespace?

File: tests/OmniSharp.Tests/Fakes/FakeLoggerFactory.cs [hunk 17]
```diff
@@ -4,13 +4,18 @@ namespace OmniSharp.Tests
 {
     public class FakeLoggerFactory : ILoggerFactory
     {
+
```
Comment (davidfowl) [inline tests/OmniSharp.Tests/Fakes/FakeLoggerFactory.cs]: extra space

File: tests/OmniSharp.Tests/project.json [hunk 18]
```diff
@@ -3,17 +3,18 @@
     "dependencies": {
         "OmniSharp": "1.0.0-*",
         "xunit": "2.1.0-beta1-*",
-        "xunit.runner.aspnet": "2.1.0-beta1-*",
-        "Microsoft.CodeAnalysis.CSharp.Workspaces": "1.0.0-beta3"
+        "xunit.runner.aspnet": "2.0.0-aspnet-beta4",
```
Comment (davidfowl) [inline tests/OmniSharp.Tests/project.json]: This isn't the one we want to use

File: .travis.yml (hunk 2 above)
Comment (Yantrio) [inline .travis.yml]: I was sleepy :(

File: tests/OmniSharp.Tests/project.json (hunk 18 above)
Comment (Yantrio) [inline tests/OmniSharp.Tests/project.json]: which is the one we want to use?

**Comment by Yantrio** [comment]
The travis builds are currently failing with an exception when publishing with the --no-source flag, do we remove this flag?

```System.IO.EndOfStreamException: Failed to read past end of stream.
  at System.IO.BinaryReader.ReadChar () [0x00000] in <filename unknown>:0 
  at Microsoft.CodeAnalysis.CvtResFile.ReadStringOrID (System.IO.BinaryReader fhIn) [0x00000] in <filename unknown>:0 
  at Microsoft.CodeAnalysis.CvtResFile.ReadResFile (System.IO.Stream stream) [0x00000] in <filename unknown>:0 
  at Microsoft.CodeAnalysis.Compilation.MakeWin32ResourceList (System.IO.Stream win32Resources, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) [0x00000] in <filename unknown>:0 ```

**Comment by davidfowl** [comment]
@Yantrio nah, downgrade to mono 3.12.x on travis. We'll need to figure this out.... somehow

**Comment by Yantrio** [comment]
but then all the tests fail :(

File: tests/OmniSharp.Tests/project.json (hunk 18 above)
Comment (willl) [inline tests/OmniSharp.Tests/project.json]: I'm guessing dnx.xunit? https://github.com/xunit/dnx.xunit

Comment (davidfowl) [inline tests/OmniSharp.Tests/project.json]: Yes

Comment (Yantrio) [inline tests/OmniSharp.Tests/project.json]: I can't find the xunit myget feed anywhere.

File: build.cmd [hunk 19]
```diff
@@ -1,31 +1,31 @@
 @echo off
 
 pushd %~dp0
-set "KRE_NUGET_API_URL=https://www.nuget.org/api/v2"
+set "DNX_NUGET_API_URL=https://www.nuget.org/api/v2"
```
Comment (davidfowl) [inline build.cmd]: Don't need this anymore

File: scripts/Omnisharp.cmd [hunk 20]
```diff
@@ -1,4 +1,3 @@
 SETLOCAL
 
-SET "KRE_APPBASE=%~dp0..\src\Omnisharp"
-%USERPROFILE%\.k\runtimes\kre-clr-win-x86.1.0.0-beta3\bin\k run %*
+%USERPROFILE%\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta4\bin\dnx %~dp0..\src\Omnisharp run %*
```
Comment (davidfowl) [inline scripts/Omnisharp.cmd]: Needs moar quotes:

```
"%USERPROFILE%\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta4\bin\dnx" "%~dp0..\src\Omnisharp" run %*
``

**Comment by Yantrio** [comment]
we seem to be blocked by https://github.com/aspnet/Home/issues/498

**Comment by nosami** [comment]
With this PR, Kestrel is started when using --stdio

**Comment by nosami** [comment]
See https://github.com/OmniSharp/omnisharp-atom/issues/341

**Comment by nosami** [comment]
Fixes the stdio not working bug :)

**Pull request closed without merging**